app/controllers/stats_controller.rb

```diff
@@ -110,4 +105,15 @@ def index
 
   end
 
+  private
+
+  def stat_element(work, element)
+    { "date": work.date,
+      "hits": work.hits,
+      "kudos.count": work.kudos.count,
+      "comment_thread_count": work.comment_thread_count,
+      "bookmarks.count": work.bookmarks.count,
+      "subscriptions.count": work.subscriptions.count,
+      "word_count": work.word_count }[element.to_sym]
```

Comment:
I think you should use a case statement here. Using a hash results in the kudos count, comment thread count, bookmarks count, and subscription count all being computed each time this function is called, even though at most one of those values is accessed and returned.

(I think some of the queries may be cached with low-level query caching, so this might not result in a 4x slowdown. But it's probably not a good idea to count on query caching to make up for this inefficiency.)